api-gateway-config/scripts/lua/routing.lua

```diff
@@ -88,33 +82,73 @@ function processCall()
   end
 end
 
---- Check redis for path parameters
--- @param red redis client instance
-function checkForPathParams(red)
-  local resourceKeys = redis.getAllResourceKeys(red, ngx.var.tenant)
+--- Find the correct redis key based on the path that's passed in
+-- @param resourceKeys list of resourceKeys to search through
+-- @param tenant tenantId
+-- @param path path to look for
+function _M.findRedisKey(resourceKeys, tenant, path)
+  -- Construct a table of redisKeys based on number of slashes in the path
+  local keyTable = {}
   for i, key in pairs(resourceKeys) do
-    local res = {string.match(key, "([^,]+):([^,]+):([^,]+)")}
-    local path = res[3] -- gatewayPath portion of redis key
-    local pathParamVars = {}
-    for w in string.gfind(path, "({%w+})") do
-      w = string.gsub(w, "{", "")
-      w = string.gsub(w, "}", "")
-      pathParamVars[#pathParamVars + 1] = w
+    local _, count = string.gsub(key, "/", "")
+    -- handle cases where resource path is "/"
+    if count == 1 and string.sub(key, -1) == "/" then
+      count = count - 1
+    end
+    count = tostring(count)
+    if keyTable[count] == nil then
+      keyTable[count] = {}
     end
-    if next(pathParamVars) ~= nil then
-      local pathPattern, count = string.gsub(path, "%{(%w*)%}", "([^,]+)")
-      local obj = {string.match(ngx.var.gatewayPath, pathPattern)}
-      if (#obj == count) then
-        for i, v in pairs(obj) do
-          ngx.ctx[pathParamVars[i]] = v
+    table.insert(keyTable[count], key)
+  end
+  -- Find the correct redisKey
+  local redisKey = utils.concatStrings({"resources:", tenant, ":", path})
+  local _, count = string.gsub(redisKey, "/", "")
+  for i = count, 0, -1 do
+    local countString = tostring(i)
+    if keyTable[countString] ~= nil then
+      for _, key in pairs(keyTable[countString]) do
+        -- Check for exact match or path parameter match
+        if key == redisKey or key == utils.concatStrings({redisKey, "/"}) or _M.pathParamMatch(key, redisKey) == true then
+          local res = {string.match(key, "([^:]+):([^:]+):([^:]+)")}
```

Comment:
This will be brittle if we ever change the way data flows into redis or allow changing the backend.